Meros.nimble

```diff
@@ -25,12 +25,69 @@ requires "stint"
 requires "nimcrypto"
 requires "normalize"
 
+# Test Dependencies
+requires "https://github.com/stefantalpalaru/nim-unittest2"
+
+
+# Procedures
+proc gatherTestFiles(dir: string): seq[string] =
+    var files: seq[string] = newSeq[string]()
+    for _, d in listDirs(dir):
+        for _, f in gatherTestFiles(d):
+            files.add(f)
+    for _, f in listFiles(dir):
+        let file: tuple[dir, name, ext: string] = splitFile(f)
+        if file.name.startsWith("Test") and file.ext == ".nim":
+            files.add(f)
+    return files
+
+proc nimbleExec(command: string) =
+    let nimbleExe: string = system.findExe("nimble")
+    if nimbleExe == "":
+        echo "Failed to find executable `nimble`."
+        quit(1)
+
+    exec nimbleExe & " " & command
+
+proc nimExec(command: string) =
+    let nimExe: string = system.findExe("nim")
+    if nimExe == "":
+        echo "Failed to find executable `nim`."
+        quit(1)
+
+    exec nimExe & " " & command
+
+# Tasks
 task clean, "Clean all build files.":
     rmDir projectDir() / "build"
 
-#Solely used to display as part of `nimble tasks`.
 task build, "Build Meros.":
     setCommand "nop"
 
 task install, "Install Meros.":
     setCommand "nop"
+
+task unit, "Run unit tests.":
+    nimbleExec "install --depsOnly"
+    # TODO: pass deps to nim based on requiresData to ensure the right dependencies are used
+    for _, f in gatherTestFiles(projectDir() / "tests"):
+        nimExec "c --outdir:" & projectDir() / "build" / "tests" & " -r " & f
+
+    # TODO: migrate `NimTests` to `tests`
+    nimExec "c -r NimTests/Test"
+
+task e2e, "Run end-to-end tests.":
+    # TODO: setup and run `PythonTests`
```

Comment:
What about renaming `PythonTests` directory to `e2e`? I think it's more understandable as to it's purpose is and the fact that they are written in python isn't really all that relevant. e.g. they can eventually be migrated to nim, go, ruby, whatever and the purpose is still the same.

Reply:
I'm not against renaming the directories, but I'd like it noted we explicitly chose not to use Nim for the E2E tests. I don't trust the Nim stdlib to debug the Nim stdlib (where the concern isn't the stdlib itself but our usage of it).

Reply:
Will take care of this in a follow-up PR.